services/BasicTypes.proto

```diff
@@ -132,7 +132,7 @@ enum TokenKycStatus {
  *
  * A key can be a "threshold key", which means a list of M keys, any N of which must sign in order for the threshold signature to be considered valid. The keys within a threshold signature may themselves be threshold signatures, to allow complex signature requirements.
  *
- * A Key can be a list of keys. Their use is dependent on context. For example, a Hedera file is created with a list of keys, where all of them must sign a transaction to create or modify the file, but only one of them is needed to sign a transaction to delete the file. So it's a single list that sometimes acts as a 1-of-M threshold key, and sometimes acts as an M-of-M threshold key.
+ * A Key can be a list of keys. Their use is dependent on context. For example, a Hedera file is created with a list of keys, where all of them must sign a transaction to create or modify the file, but only one of them is needed to sign a transaction to delete the file. So it's a single list that sometimes acts as a 1-of-M threshold key, and sometimes acts as an M-of-M threshold key. A key list is always an M-of-M, unless specified otherwise in documentation. A key list can have nested key lists or threshold keys. Nested key lists are always M-of-M. A key list can have repeated Ed25519 public keys and a single signature for a repeated public key will represent each of the repeated public keys. For example, a key list has a 3-of-3 key. Two of those keys are the same public key x. When a user signs with key x, the key list will have met the requirement for 2 out of the 3 required signatures.
```

Comment:
Little nit: I generally don't like examples in definitions, but that's not a big issue

Reply:
Suggested Update: A key can be a "key list", which means a list of M keys which all keys must sign, unless specified otherwise for a specific transaction type (e.g. FileDeleteTransactionBody), in order for the the keylist signature to be considered valid. The keys within a keylist signature may contain both keylist and threshold keys which required their threshold signatures to be meet to be considered valid. A keylist may contain repeated keys but all repeated keys are only required to sign once.